lib/rules/no-warning-comments.js

```diff
@@ -141,12 +143,25 @@ module.exports = {
 
             const matches = commentContainsWarningTerm(node.value);
 
-            matches.forEach(matchedTerm => {
+            matches.forEach(({ type, comment }) => {
+                let commentsToDisplay = "";
+                let truncated = false;
+
+                for (const c of comment.trim().split(/\s+/u)) {
+                    commentsToDisplay = commentsToDisplay.trim();
+                    if (c.length + commentsToDisplay.length <= CHAR_LIMIT) {
```

Comment:
This can be off-by-one.

Maybe we could first make the next string, then check its length and if it's okay assign it to `commentToDisplay`.

Reply:
like this using a temporary variable ?

```js
for (const c of comment.trim().split(/\s+/u)) {
  commentToDisplay = commentToDisplay.trim();
  const tmp = `${commentToDisplay} ${c}`;

  if (tmp.length <= CHAR_LIMIT) {
    commentToDisplay = tmp;
  } else {
    truncated = true;
  }
}
```

Reply:
Yes, something like that

Reply:
if we construct the next string first as mentioned above in the snippet, there may be cases where the `...` is being appended with a nonfull word i.e

instead of `and...`, there is a chance of having like `a...` as well in this [line](https://github.com/eslint/eslint/pull/13522/files#diff-0e787f8a02ecbdc84bc736003f9736ceR230)

I can refactor as this i.e extracting the condition to a variable. 

```js
for (const c of comment.trim().split(/\s+/u)) {
  commentToDisplay = commentToDisplay.trim();
  const len = commentToDisplay.length + c.length;

  if (len <= CHAR_LIMIT) {
    commentToDisplay = `${commentToDisplay} ${c}`;
  } else {
    truncated = true;
  }
}
```
WDYT ?

Reply:
> if we construct the next string first as mentioned above in the snippet, there may be cases where the `...` is being appended with a nonfull word i.e
> 
> instead of `and...`, there is a chance of having like `a...` as well in this [line](https://github.com/eslint/eslint/pull/13522/files#diff-0e787f8a02ecbdc84bc736003f9736ceR230)

Hm, I'm not sure how could that happen if we're always appending full words

Reply:
yeah , 

![image](https://user-images.githubusercontent.com/26347874/89787988-154b5280-db3c-11ea-9e98-edeaba98e04a.png)

Reply:
That's probably the full word "a" that appears later in the comment, not the first letter of "and".

We should break the loop after `truncated = true`.

Reply:
done